blocksuite/affine/shared/src/viewport-renderer/renderer-utils.ts

```diff
@@ -23,58 +27,104 @@ export function syncCanvasSize(canvas: HTMLCanvasElement, host: HTMLElement) {
   canvas.style.pointerEvents = 'none';
 }
 
+function clientToModelCoord(
+  viewport: ViewportRecord,
+  clientCoord: [number, number]
+) {
+  const { left, top, viewportX, viewportY, zoom, viewScale } = viewport;
+
+  const [clientX, clientY] = clientCoord;
+  const viewportInternalX = clientX - left;
+  const viewportInternalY = clientY - top;
+  const modelX = viewportX + viewportInternalX / zoom / viewScale;
+  const modelY = viewportY + viewportInternalY / zoom / viewScale;
+  return [modelX, modelY];
+}
```

Comment:
I think we should move this function to gfx/viewport.ts and modify Viewport.toModelCoordFromClientCoord to ensure they have the same implementation.🤔

Reply:
Great advice, thanks!